arches/app/views/search.py

```diff
@@ -348,40 +348,37 @@ def get_dsl_from_search_string(request):
 
 
 def search_results(request, returnDsl=False):
-    se = SearchEngineFactory().create()
     search_filter_factory = SearchFilterFactory(request)
     searchview_component_instance = search_filter_factory.get_searchview_instance()
     if not searchview_component_instance:
-        unavailable_core_name = search_filter_factory.get_searchview_name()
-        ret = {
-            "success": False,
-            "message": _("No core-search component named {0}").format(
-                unavailable_core_name
-            ),
-        }
-        return JSONResponse(ret, status=406)
-
-    search_query_object = {"query": Query(se)}
-    response_object = {"results": None}
+        unavailable_searchview_name = search_filter_factory.get_searchview_name()
+        message = _("No search-view named {0}").format(unavailable_searchview_name)
+        return JSONErrorResponse(
+            _("Search Failed"),
+            message,
+            status=400,
+        )
 
     response_object, search_query_object = (
         searchview_component_instance.handle_search_results_query(
-            search_query_object, response_object, search_filter_factory, returnDsl
+            search_filter_factory, returnDsl
         )
     )
-
-    if returnDsl:
-        return search_query_object.pop("query", None)
-
-    if response_object:
+    if isinstance(response_object, str):
         return JSONResponse(content=response_object)
-
+    elif not response_object.get("success", False):
+        message = response_object.get(
+            "message", _("There was an error retrieving the search results")
+        )
+        return JSONErrorResponse(
+            _("Search Failed"),
+            message,
+            status=500,
+        )
```

Comment:
```suggestion
    try:
        response_object, search_query_object = (
            searchview_component_instance.handle_search_results_query(
                search_filter_factory, returnDsl
            )
        )
        if returnDsl:
            return search_query_object.pop("query")
        else:
            return JSONResponse(content=response_object)
    except Exception as e:
        message = e.args[0].get(
            "message", _("There was an error retrieving the search results")
        )
        return JSONErrorResponse(
            _("Search Failed"),
            message,
            status=500,
        )
```

Reply:
I'm getting a "Applying suggestions on deleted lines is currently not supported." error on github when I try to apply this commit via the github GUI so I'm committing this change manually.